### PR TITLE
cleanup PMacc's event system

### PR DESCRIPTION
**Pull request by psychocoderHPC**
The motivation behind this PR is to reduce the usage of macros and avoid problems with cyclic includes.
In the past, I refactored PMacc so that we can access all manager singletons in PMacc via the god object `Environment`.
I know now that this was not a good idea because it introduces a lot of issues with cyclic includes and avoids separating non-template PMacc interfaces in `hpp`-files and implementations.

- remove macros `__startTransaction(...), ...`
  - `eventSystem.hpp` is providing for the user all necessary entry functions for the event system
- separate more classes into hpp,cpp files remove `Manager()`, `TransactionManager()` singleton access from the god object `Environment`

This PR should not affect any user input of PIConGPU, only the developer must take care that you need to use the interfaces in `pmacc/eventSystem/eventSystem.hpp` to interact with the event system of PMacc.

### CODE REVIEW

File: include/pmacc/Environment.def [hunk 1]
```diff
@@ -32,5 +33,100 @@ namespace pmacc
     {
         struct Environment;
 
+        /** collect state variables of the environment context
+         *
+         * This class handle the initialization and finalize of the
+         * MPI context and the selection of the GPU.
+         */
+        class EnvironmentContext
```
Comment (psychocoderHPC) [inline include/pmacc/Environment.def:41]: This class is moved into the `def` file because it is required to take care that the transaction system is not started before we selected a device.
Moving this class here is no problem because it is not pulling files with include side effects.

File: include/pmacc/particles/tasks/TaskParticlesSend.hpp [hunk 2]
```diff
@@ -22,7 +22,8 @@
 #pragma once
 
 #include "pmacc/Environment.hpp"
-#include "pmacc/eventSystem/EventSystem.hpp"
+#include "pmacc/eventSystem/Manager.hpp"
+#include "pmacc/eventSystem/tasks/MPITask.hpp"
```
Comment (PrometheusPi) [inline include/pmacc/particles/tasks/TaskParticlesSend.hpp:26]: @psychocoderHPC  What was added in the code that now needs `MPITask.hpp`

Comment (psychocoderHPC) [reply include/pmacc/particles/tasks/TaskParticlesSend.hpp:26]: before it was transitively included via `EventSystem.hpp`

File: include/pmacc/eventSystem/streams/EventStream.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2013-2022 Heiko Burau, Rene Widera, Alexander Grund
+/* Copyright 2013-2022 Felix Schmitt, Rene Widera, Benjamin Worpitz
```
Comment (PrometheusPi) [inline include/pmacc/eventSystem/streams/EventStream.cpp:1]: What happened here with the authors?

File: include/pmacc/eventSystem/streams/EventStream.cpp [hunk 4]
```diff
@@ -19,12 +19,32 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
-#pragma once
```
Comment (PrometheusPi) [inline include/pmacc/eventSystem/streams/EventStream.cpp:22]: Why did you remove the `#pragma once`?

**Review by PrometheusPi — APPROVED** [review]

File: include/pmacc/eventSystem/streams/EventStream.cpp (hunk 3 above)
Comment (psychocoderHPC) [reply include/pmacc/eventSystem/streams/EventStream.cpp:1]: The files got removed but github tries to link files and is thinking we renamed the file.
So this change is OK.

File: include/pmacc/eventSystem/streams/EventStream.cpp (hunk 4 above)
Comment (psychocoderHPC) [reply include/pmacc/eventSystem/streams/EventStream.cpp:22]: In cpp files you can not have pragma once. This file is new but github is thinking the file is moved.
